ts/components/services/TosAndPrivacyBox.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import { StyleSheet, View } from "react-native";
+
+import { TranslationKeys } from "../../../locales/locales";
+import I18n from "../../i18n";
+import { handleItemOnPress } from "../../utils/url";
+
+import IconFont from "../ui/IconFont";
+import { H3 } from "../core/typography/H3";
+import { Link } from "../core/typography/Link";
+import { IOColors } from "../core/variables/IOColors";
+import ItemSeparatorComponent from "../ItemSeparatorComponent";
+
+const styles = StyleSheet.create({
+  infoItem: {
+    paddingVertical: 16,
+    flex: 1
+  },
+  header: {
+    flexDirection: "row",
+    paddingVertical: 8
+  },
+  icon: {
+    marginRight: 3.85
+  }
+});
+
+type Props = {
+  privacyUrl?: string;
+  tosUrl?: string;
+};
+
+/**
+ * Renders a dedicated section with TOS, Privacy urls, and the header.
+ */
+const TosAndPrivacy: React.FC<Props> = ({ tosUrl, privacyUrl }) => (
+  <View>
+    <View style={styles.header}>
+      <IconFont
+        name={"io-lucchetto"}
+        color={IOColors.bluegrey}
+        style={styles.icon}
+      />
+      <H3 weight={"SemiBold"} color={"bluegrey"}>
+        {I18n.t("services.tosAndPrivacy")}
+      </H3>
+    </View>
+    {tosUrl && (
+      <>
+        {renderLinkRow("services.tosLink", tosUrl)}
+        <ItemSeparatorComponent noPadded />
+      </>
+    )}
+    {privacyUrl && (
+      <>
+        {renderLinkRow("services.privacyLink", privacyUrl)}
+        <ItemSeparatorComponent noPadded />
+      </>
+    )}
+    <ItemSeparatorComponent noPadded />
+  </View>
+);
+
+const renderLinkRow = (text: TranslationKeys, href: string) => (
+  <View style={styles.infoItem}>
+    <Link
+      ellipsizeMode={"tail"}
+      onPress={handleItemOnPress(href)}
```

Comment:
```suggestion
      onPress={() =>
        openWebUrl(href, () => showToast(I18n.t("global.jserror.title")))
      }
```
Since the component has been refactored, we can also avoid using `handleItemOnPress` (we are always trying to open a web url and not a custom url) and use the more safe `openWebUrl` that is able to handle even the cases of malformed url and provides a feedback to the user (the link we are trying to open comes from remote data and we do not have the certainty that it is valid).

https://user-images.githubusercontent.com/26501317/124605661-707ce680-de6c-11eb-9e5f-3f560061615c.mov

This is the effect if I'm trying to open a malformed url with this suggestion. Do you think could be an improvement?

PS: the error message used in the toast is just a generic message, used for placeholder.

Reply:
Updated [here](https://github.com/pagopa/io-app/pull/3200/commits/3ef427e3ae046af2cb180cc1a0fa231a59c5a767#diff-a6c4bb541039dee54bf32f6b1bb41ed1970b3b733f22099e3cb609b9f8c528b4R38) and tested [here](https://github.com/pagopa/io-app/pull/3200/commits/3ef427e3ae046af2cb180cc1a0fa231a59c5a767#diff-b61fe31b0909f945faba5f3797accd2d2ec9404e8d16f32f4adcadb1706e42d8R64)